### PR TITLE
Implemented serialized data interface

### PR DESCRIPTION
**Pull request by jmadden173**
Most of the code is in `proto/` with `c` and `python` libraries along with tests. Let me know if there is enough documentation available to use both libraries.

### CODE REVIEW

File: proto/c/include/transcoder.h
Comment (stevegtaylor) [inline proto/c/include/transcoder.h:1]: File brief could use more detail

Comment (stevegtaylor) [inline proto/c/include/transcoder.h:1]: encode terosmeasurments brief says encode power measurment

File: proto/python/src/soil_power_sensor_protobuf/transforms.py
[the remark this answers is not in the document]
Comment (stevegtaylor) [reply proto/python/src/soil_power_sensor_protobuf/transforms.py:1]: Please change to doxygen

File: stm32/test/test_transcoder/test_transcoder.c
Comment (stevegtaylor) [inline stm32/test/test_transcoder/test_transcoder.c:1]: Add description to systemclock_config

Comment (stevegtaylor) [inline stm32/test/test_transcoder/test_transcoder.c:1]: Where did you get the testing values?

Comment (stevegtaylor) [inline stm32/test/test_transcoder/test_transcoder.c:1]: More documentation

**Review by stevegtaylor — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-python.yaml
[the remark this answers is not in the document]
Comment (jmadden173) [reply .github/workflows/publish-python.yaml:1]: TODO: Add comment linking to boilerplate code

**Comment by jmadden173** [comment]
@stevegtaylor Updated the documentation and resolved all comments. As a last check can you try running the STM32 tests to check that they work and are configured correctly?

For the CI checks, the publish to TestPyPI failed because it has already been upload. For the ESP32, there is not source files yet therefore fails. Both of these are fine an cannot be fixed in this PR.

**Comment by jmadden173** [comment]
@stevegtaylor Update on this?

**Comment by stevegtaylor** [comment]
> @stevegtaylor Update on this?

Working on openocd right now

**Comment by stevegtaylor** [comment]
4 tests passed, with the latest code. Had to play around with the location of tool_openocd.py.

**Review by stevegtaylor — APPROVED** [review]

**Pull request merged**